src/avatar.jsx

```diff
@@ -156,7 +154,7 @@ const Avatar = React.createClass({
           className={className}
         >
           {icon && React.cloneElement(icon, {
-            color: color,
+            color: color || avatar.color,
             style: Object.assign(styles.icon, icon.props.style),
```

Comment:
Does that work for all icon types, like SVG?  Do you also need to provide a fill (I haven't checked the code, maybe not)?